lib/MockXMLHttpRequest.js

```diff
@@ -171,6 +165,8 @@ MockXMLHttpRequest.prototype.send = function(data) {
         }, typeof(timeout) === 'number' ? timeout : self.timeout+1);
 
       } else {
+        //trigger a load event to indicate the data has been sent
+        self.upload.trigger('load');
```

Comment:
possibly so.  I didn't need any of the other events to get my tests working, but having a full lifecycle would probably be useful.